sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,158 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import io.sentry.TypeCheckHint
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECTION_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECT_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.SECURE_CONNECT_EVENT
+import io.sentry.util.UrlUtils
+import okhttp3.Call
+import okhttp3.Response
+import java.net.URL
+
+private val uuidRegex by lazy { Regex("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}") }
+
+internal class SentryOkHttpEvent(private val hub: IHub, private val call: Call) {
+    private val eventSpans: MutableMap<String, ISpan> = HashMap()
+    private val breadcrumb: Breadcrumb
+    internal val callRootSpan: ISpan?
+    private var response: Response? = null
+
+    init {
+        val urlDetails = UrlUtils.parse(call.request().url.toString())
+        val url = urlDetails.urlOrFallback
+        val trimmedUrl: String = trimUrl(url)
+        val host: String = call.request().url.host
+        val encodedPath: String = call.request().url.encodedPath
+        val method: String = call.request().method
+
+        // We start the call span that will contain all the others
+        callRootSpan = hub.span?.startChild("http.client", "$method $url")
+
+        urlDetails.applyToSpan(callRootSpan)
+
+        // We setup a breadcrumb with all meaningful data
+        breadcrumb = Breadcrumb.http(url, method)
+        breadcrumb.setData("url", url)
+        breadcrumb.setData("filtered_url", trimmedUrl)
+        breadcrumb.setData("host", host)
+        breadcrumb.setData("path", encodedPath)
+        breadcrumb.setData("http.method", method)
```

Comment:
`m`: should we just change the key here instead? Or is Sentry frontend not aware of this key and won't properly format the description? https://github.com/getsentry/sentry-java/blob/e9bec6ec2c5092673e12b21b008c3abf8e3cccc1/sentry/src/main/java/io/sentry/Breadcrumb.java#L163

Reply:
Also, just my personal preference, but I'm not sure if this looks aligned - I guess I would either prepend every key with `http.` here or just remove the `http.` prefix altogether (we don't have breadcrumb data conventions I believe). Do you think that makes sense?
![image](https://github.com/getsentry/sentry-java/assets/4999776/f00a0158-0549-474c-b100-4283906c928f)

Reply:
data convention should apply to span only, so we can remove the `http.` prefix, to make it consistent to the previous implementation (the `SentryOkHttpInterceptor`)